common/utilities/number/stabilized-value.h

```diff
@@ -0,0 +1,119 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <string>
+#include <atomic>
+#include <mutex>
+
+namespace utilities {
+namespace number {
+
+// The stabilized value implement a history based stable value.
+// It supply a way to define and get values with a required stabilization.
+// When the user adds a value, the value is inserted into the history
+// When the user call "get" the last stable value is returned
+// Inputs: Template <value_type>
+//         required history size
+//         required stabilization percentage as a fraction , range [0-1]
+// The stabilized_value class is thread safe.
+template < typename T > class stabilized_value
+{
+public:
+    stabilized_value( size_t history_size )
+        : _history_size( history_size )
+        , _last_stable_value( 0 )
+    {
+
+    }
+
+    stabilized_value() = delete;
+    stabilized_value(const stabilized_value &) = delete;
+
+
+    void add( T val )
+    {
+        const std::lock_guard< std::mutex > lock( _mutex );
+        if( _values.empty() )
+        {
+            _values.push_back( val );
+            _last_stable_value = val;
+            return;
```

Comment:
The value of `_recalc_stable_val` is unknown here. I think you can set it here.

Reply:
It was because it was set at the Ctor and on the reset function. the only ways you get a clear history. Now we need to add it here to after the other comments..

Reply:
Done